packages/app/components/stackbox/FrequencyOptionsCard.tsx

```diff
@@ -0,0 +1,130 @@
+    case FREQUENCY_OPTIONS.week:
+      return add(new Date(), { weeks: timeAmount });
+    case FREQUENCY_OPTIONS.month:
+      return add(new Date(), { months: timeAmount });
+    default:
+      throw new Error("Invalid frequency option");
+  }
+};
+
+const getCroppedFrequency = (frequency: FREQUENCY_OPTIONS) => {
+  const isMonthFrequency = frequency === FREQUENCY_OPTIONS.month;
+
+  return isMonthFrequency ? frequency.substring(0, 2) : frequency.charAt(0);
+};
+
+export const FrequencyOptionsCard = ({
+  frequency,
+  setEndDate,
+}: FrequencyOptionsCardProps) => {
+  const [defaultFrequency, setDefaultFrequency] = useState(
+    defaultFrequencyOptions[frequency][0]
+  );
+  const [customFrequency, setCustomFrequency] = useState("");
+
+  const handleCustomFrequencyChange = (
+    event: ChangeEvent<HTMLInputElement>
+  ) => {
+    const newValue = event.target.value;
+    const isWithinRange = Number(newValue) <= maxCustomFrequencies[frequency];
+
+    if (postiveIntegerOnly.test(newValue) && isWithinRange) {
+      setDefaultFrequency("");
+      setCustomFrequency(newValue);
+    }
+  };
+
+  useEffect(() => {
+    setDefaultFrequency(defaultFrequencyOptions[frequency][0]);
+  }, [frequency]);
+
+  useEffect(() => {
+    const newFrequency = customFrequency ? customFrequency : defaultFrequency;
+
+    setEndDate(getEndDateByDefaultFrequency(frequency, Number(newFrequency)));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [defaultFrequency, customFrequency]);
+
+  const placeholder = `Custom ${frequency} (max: ${maxCustomFrequencies[frequency]})`;
+
+  return (
+    <div
+      className={twMerge([
+        "flex flex-col p-3 rounded-2xl gap-1",
+        "bg-surface-25 border border-surface-50",
+      ])}
+    >
+      <BodyText className="text-em-low">Duration</BodyText>
+      <div className="flex gap-2 flex-col md:flex-row">
+        <div className="flex gap-2">
+          {defaultFrequencyOptions[frequency].map((freqOption) => {
+            const isSelected = defaultFrequency === freqOption;
+
+            return (
+              <RadioButton
+                checked={isSelected}
+                className="min-w-[70px] h-10"
+                id={freqOption}
+                key={freqOption}
+                name={freqOption}
+                onChange={(event) => {
+                  setDefaultFrequency(event.target.value as FREQUENCY_OPTIONS);
+                }}
+                value={freqOption}
+              >
+                <BodyText className={!isSelected ? "text-em-med" : ""} size={2}>
+                  {`${freqOption} ${getCroppedFrequency(frequency)}`}
+                </BodyText>
+              </RadioButton>
+            );
+          })}
+        </div>
+        <TextInput
+          id="custom-frequency-option"
+          min={1}
+          onChange={handleCustomFrequencyChange}
+          placeholder={placeholder}
+          value={customFrequency}
+        />
```

Comment:
We are allowing 0. Maybe we shouldn't allow 0 or empty. If user has 0/empty it defaults to the frequency default value.